CMakeLists.txt

```diff
@@ -163,6 +163,13 @@ ADD_CUSTOM_TARGET(setup_tests
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
   COMMENT "Enabling all tests ...")
 
+# Disable the ability to run all tests in an in-source build:
+IF (("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}") AND ASPECT_RUN_ALL_TESTS)
+  SET(ASPECT_RUN_ALL_TESTS OFF CACHE BOOL "" FORCE)
+  MESSAGE(FATAL_ERROR "\nEnabling all tests is not supported in in-source builds. Please create a separate build directory!\n")
```

Comment:
That's fine too, but isn't the SET a dead store in that case because we're about to die anyway?

Reply:
No. Otherwise you end up in a state where the variable is set and cmake fails when you run it until you figure out how to fix it (``cmake -D ASPECT_RUN_ALL_TESTS=OFF``).

Reply:
Ah, the crazy cache...